packages/aws-rfdk/lib/core/test/deployment-instance.test.ts

```diff
@@ -0,0 +1,763 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  arrayWith,
+  countResources,
+  expect as expectCDK,
+  haveResourceLike,
+  ResourcePart,
+  stringLike,
+} from '@aws-cdk/assert';
+import { CfnLaunchConfiguration } from '@aws-cdk/aws-autoscaling';
+import {
+  AmazonLinuxImage,
+  ExecuteFileOptions,
+  InstanceClass,
+  InstanceSize,
+  InstanceType,
+  SecurityGroup,
+  SubnetType,
+  Vpc,
+} from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import { ILogGroup } from '@aws-cdk/aws-logs';
+import { Bucket } from '@aws-cdk/aws-s3';
+import { Asset } from '@aws-cdk/aws-s3-assets';
+import { StringParameter } from '@aws-cdk/aws-ssm';
+import * as cdk from '@aws-cdk/core';
+import { Duration } from '@aws-cdk/core';
+
+import {
+  DeploymentInstance,
+  DeploymentInstanceProps,
+} from '../lib/deployment-instance';
+import { resourceTagMatcher, testConstructTags } from './tag-helpers';
+
+
+const DEFAULT_CONSTRUCT_ID = 'DeploymentInstance';
+
+/**
+ * Machine image that spies on the following user data methods:
+ *
+ * * `.addOnExitCommands`
+ * * `.addExecuteFileCommand`
+ */
+class AmazonLinuxWithUserDataSpy extends AmazonLinuxImage {
+  public getImage(scope: cdk.Construct) {
+    const result = super.getImage(scope);
+    jest.spyOn(result.userData, 'addOnExitCommands');
+    jest.spyOn(result.userData, 'addExecuteFileCommand');
+    return result;
+  }
+}
+
+describe('DeploymentInstance', () => {
+  let app: cdk.App;
+  let depStack: cdk.Stack;
+  let stack: cdk.Stack;
+  let vpc: Vpc;
+  let target: DeploymentInstance;
+  let deploymentInstanceProps: DeploymentInstanceProps;
+
+  beforeEach(() => {
```

Comment:
Running this setup method before each test and having tests only assert on one field seems like a heavyweight way to do testing. Is there any opportunity to switch some of these `beforeEach()` setup methods to `beforeAll()` methods?